src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs

```diff
@@ -10,16 +10,35 @@
 
 namespace System.Text.Encodings.Web
 {
+    /// <summary>
+    /// Represents a type used to do HTML encoding.
+    /// </summary>
     public abstract class HtmlEncoder : TextEncoder
     {
+        /// <summary>
+        /// Returns default built-in instance of HtmlEncoder.
+        /// </summary>
         public static HtmlEncoder Default
         {
             get { return DefaultHtmlEncoder.Singleton; }
         }
+
+        /// <summary>
+        /// Creates a new instalce of HtmlEncoder with provided settings.
+        /// </summary>
+        /// <param name="settings">Settings used mainly to specify which characters end up being encoded.</param>
+        /// <returns>Returns new instance of the encoder.</returns>
         public static HtmlEncoder Create(TextEncoderSettings settings)
         {
             return new DefaultHtmlEncoder(settings);
         }
+
+        /// <summary>
+        /// Creates a new instalce of HtmlEncoder specifying character to be encoded.
+        /// </summary>
+        /// <param name="allowedRanges">Set of characters that the endocer is allowed to not encode.</param>
+        /// <returns>Returns new instance of the encoder.</returns>
+        /// <remarks>Some characters in the ranges might still get encoded, i.e. this parameter is just a hint.</remarks> 
```

Comment:
This seems like strange behavior.  Could you elaborate with an example to help clarify? e.g. "This parameter is just a hint, and the encoder encode characters in these ranges.  For example, ...".